src/MusicPlaylist.java

```diff
@@ -0,0 +1,94 @@
+import components.queue.Queue;
+import components.queue.Queue1L;
+
+/**
+ * {@code MusicPlaylist} represented as {@Queue<String>} with implementations of
+ * primary methods.
+ *
+ * @convention <pre>
+ *[the key of $this.rep is the song title and the value of $this.rep is the artist of the song]
+ * </pre>
+ *
+ * @correspondnece <pre>
+ * </pre>
+ *
+ * @author Jin Yang
+ *
+ */
+
+public class MusicPlaylist extends MusicPlaylistSecondary {
+
+    /**
+     * Representation of {@code this}.
+     *
+     */
+    private Queue rep;
+
+    /**
+     * Creator of inital representation.
+     */
+    private void createNewRep() {
+        this.rep = new Queue1L<String>();
```

Comment:
This is perfectly fine. If you want something a bit more flexible, I might recommend that you store Maps or Songs in your queue. Maps can be used to store pairs of info like: `{(artist, ...), (title, ...), (pathToSong, ...), (pathToCoverArt, ...)}`. Alternatively, you could make your own inner Song class like this:
```java
class Song {
  public String title;
  public String artist;
}
```
If you do the latter, you will have a little more work on your hands with inner interfaces.